src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodTranslator.cs

```diff
@@ -50,17 +51,29 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
                 (!arguments[0].Type.IsGenericType || arguments[0].Type.GetGenericTypeDefinition() != typeof(List<>)))
                 return null;
```

Comment:
Overlooked this. You duplicated the logic below, so no need in this except argument count check.

Reply:
You're right, that was the original idea.